src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -118,6 +118,52 @@ sealed class NumericRange[T](
     }
   }
 
+  override final def indexOf[B >: T](elem: B, from: Int = 0): Int = {
+    elem match {
+      case i: T =>
```

Comment:
Unfortunately, that's already a no-go. `T` is erased, so this type test "always succeeds" but will crash later one in unpredictable ways.

Reply:
Yes. I just noticed the compiler failing on this. Very unfortunate.